src/sentry/api/endpoints/project_metrics_extraction_rules.py

```diff
@@ -81,62 +88,86 @@ def post(self, request: Request, project: Project) -> Response:
+        if not config_update:
             return Response(status=200)
 
         try:
-            state_update = self._generate_updated_rule_objects(rules_update)
-            persisted_rules = update_metrics_extraction_rules(project, state_update)
-            updated_rules = serialize(
-                persisted_rules, request.user, MetricsExtractionRuleSerializer()
-            )
-        except Exception as e:
-            return Response(status=400, data={"detail": str(e)})
-
-        return Response(status=200, data=updated_rules)
-
-    def _generate_updated_rule_objects(
-        self, updated_rules: list[dict[str, Any]]
-    ) -> dict[str, MetricsExtractionRule]:
-        state_update = {}
-        for updated_rule in updated_rules:
-            rule = MetricsExtractionRule.from_dict(updated_rule)
-            mri = rule.generate_mri()
-            state_update[mri] = rule
-
-        return state_update
-
-    def _generate_deleted_rule_objects(
-        self, updated_rules: list[dict[str, Any]]
-    ) -> Sequence[MetricsExtractionRule]:
-        state_update = []
-        for updated_rule in updated_rules:
-            state_update.append(MetricsExtractionRule.from_dict(updated_rule))
+            with transaction.atomic(router.db_for_write(SpanAttributeExtractionRuleConfig)):
+                configs = []
+                for obj in config_update:
+                    config = SpanAttributeExtractionRuleConfig.objects.get(
+                        project=project, span_attribute=obj["spanAttribute"]
+                    )
+                    config.aggregates = obj["aggregates"]
+                    config.unit = HARD_CODED_UNITS.get(obj["spanAttribute"], obj["unit"])
+                    config.tags = obj["tags"]
+                    config.save()
+                    config.refresh_from_db()
+                    configs.append(config)
+
+                    # delete conditions not present in update
+                    included_conditions = [x["id"] for x in obj["conditions"]]
+                    SpanAttributeExtractionRuleCondition.objects.filter(config=config).exclude(
+                        id__in=included_conditions
+                    ).delete()
+
+                    for condition in obj["conditions"]:
+                        condition_id = condition["id"] if "id" in condition else None
+                        SpanAttributeExtractionRuleCondition.objects.update_or_create(
+                            id=condition_id,
+                            config=config,
+                            defaults={
+                                "value": condition["value"],
+                                "created_by_id": request.user.id,
+                            },
+                        )
+                schedule_invalidate_project_config(
+                    project_id=project.id, trigger="span_attribute_extraction_configs"
+                )
+
+        except Exception:
+            return Response(status=400)
```

Comment:
Should we capture errors here as well, or are all errors invalid user input?

Reply:
Added a call to capture_exception()